crates/bevy_reflect/src/array.rs

```diff
@@ -400,8 +391,6 @@ impl<'a> IntoIterator for &'a DynamicArray {
 }
 
 impl_type_path!((in bevy_reflect) DynamicArray);
-#[cfg(feature = "functions")]
-crate::func::macros::impl_function_traits!(DynamicArray);
 
```

Comment:
oh this is just git doing git things! i'll add them back when i get a moment!

Reply:
actually i'm completely mistaken! there were some git issues with one of the merges, but this was actually intentional.

i'm not too familiar with all the new function reflection stuff yet, so correct me if i'm wrong, but i think we don't want to implement the relevant traits for these dynamic types, because we don't want users to try downcasting to them (because they probably want to be using some concrete type).

Reply:
Yeah I was debating on something like this when working on it. Specifically, I was thinking we could maybe use `FromReflect` to auto-convert dynamic arguments to their concrete type, but—aside from limiting arguments to `FromReflect` types—it meant that users couldn't have their functions explicitly take in a dynamic type argument.

I don't think it should break anything to remove these, but I don't know if we should for the same reason: users might want to explicitly take `DynamicStruct` or something as an argument.

Is the issue here that they can't be downcast due to not being `Reflect`? If so, then I think we can remove it for now and explore potential fixes in the future. It's not worth blocking this PR over.

Reply:
yeah internally the macros use `take_ref` etc. for downcasting (which in turn calls `try_downcast_ref` etc.) which all bind `T: Reflect`.

Reply:
Gotcha. Then yeah let's just remove these for now. Thanks for the info!